packages/react-aria-components/src/Group.tsx

```diff
@@ -69,7 +69,8 @@ export const Group = /*#__PURE__*/ (forwardRef as forwardRefType)(function Group
 
   let {hoverProps, isHovered} = useHover({onHoverStart, onHoverChange, onHoverEnd, isDisabled});
   let {isFocused, isFocusVisible, focusProps} = useFocusRing({
-    within: true
+    within: true,
+    isTextInput: true
```

Comment:
The reason for the focus ring always reappearing as you type in the search field is because the Group itself has `focusRing` applied to it but doesn't have `isTextInput`. When `useFocusVisible` detects the native keyboards we dispatch on the autocomplete collection/collection items, it believes that it needs to update the modality for the Group. Normally this would be correct, however, the Group is still focused despite those events happening on different elements and thus its focus visibility get updated. Still trying to think of a good way to handle this